### PR TITLE
Improve system tests descriptions

### PR DESCRIPTION
**Pull request by laoneo**
Unifies the test descriptions. The describe text should always be:

`Test that the {component name} {view name} view`

While the description of the test itself should be a continuation of the description text like:

`can {the tests which are performed}`

Both texts should be possible to concat so a meaningful sentence will be built. Because these texts will be displayed in cypress like:
image

### CODE REVIEW

**Comment by laoneo** [comment]
@brianteeman can you have a look here if I screwed up something?

**Comment by brianteeman** [comment]
@laoneo lol - I was going to take a look at all of this today. Looks like there might still be a few tweaks needed. setting up my pc for tests and then will post back shortly

File: tests/System/integration/site/components/com_contact/Categories.cy.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
-describe('Test that the Categories of com_contact ', () => {
-  it('can display a list of categories of contacts in menu item', () => {
+describe('Test that the contact categories view', () => {
```
Comment (sandewt) [inline tests/System/integration/site/components/com_contact/Categories.cy.js]: Distinguish between 'frontend' and 'backend':
`Test in frontend that the contact categories view`

File: tests/System/integration/site/components/com_users/Remind.cy.js [hunk 2]
```diff
@@ -13,11 +12,10 @@ describe('Test the remind view', () => {
       });
   });
 
-  it('can display the remind view for test user', () => {
+  it('can send a reminder mail for a test user without a menu item', () => {
```
Comment (sandewt) [inline tests/System/integration/site/components/com_users/Remind.cy.js]: `email` , see https://manual.joomla.org/docs/user-interface-text/words2watch/

Comment (laoneo) [reply tests/System/integration/site/components/com_users/Remind.cy.js]: done

File: tests/System/integration/site/components/com_contact/Categories.cy.js (hunk 1 above)
Comment (laoneo) [reply tests/System/integration/site/components/com_contact/Categories.cy.js]: done

**Comment by laoneo** [comment]
@sandewt thanks for the hint, changed it

File: tests/System/integration/administrator/components/com_categories/Categories.cy.js [hunk 3]
```diff
@@ -1,4 +1,4 @@
-describe('Test that the categories back end list', () => {
+describe('Test in backend that the categories back end list', () => {
```
Comment (brianteeman) [inline tests/System/integration/administrator/components/com_categories/Categories.cy.js]: does it really need to say backend twice
How about
Test in the administrator that the categories list
Test in the site that the categories list

Comment (laoneo) [reply tests/System/integration/administrator/components/com_categories/Categories.cy.js]: I just removed the second "back end" part of all strings

File: configuration.php.bak [hunk 4]
```diff
@@ -0,0 +1,121 @@
+<?php
```
Comment (brianteeman) [inline configuration.php.bak]: did you intent to commit this file?

Comment (laoneo) [reply configuration.php.bak]: not really

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on 41e01f19a093cf7c6215fb7cee26a52172d331c5<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40386">issues.joomla.org/tracker/joomla-cms/40386</a>.</sub>

**Comment by brianteeman** [comment]
These text patterns should be documented<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40386">issues.joomla.org/tracker/joomla-cms/40386</a>.</sub>

File: tests/System/integration/site/components/com_users/Reset.cy.js [hunk 5]
```diff
@@ -1,22 +1,20 @@
-describe('Test the reset view', () => {
-  it('can display the reset view for test user', () => {
+describe('Test in frontend that the users reset view', () => {
+  it('can send out a reset email for a test user without a menu item', () => {
```
Comment (sandewt) [inline tests/System/integration/site/components/com_users/Reset.cy.js]: send out = send

File: tests/System/integration/site/components/com_users/Reset.cy.js [hunk 6]
```diff
@@ -1,22 +1,20 @@
...
-
         cy.get('#jform_email').type('test@example.com');
         cy.get('.controls > .btn').click();
 
         cy.get('#system-message-container').should('contain.text', 'If the email address you entered is registered on this site you will shortly receive an email with a link to reset the password for your account.');
       });
   });
 
-  it('testing reset form for test user through menu item', () => {
+  it('can send out a reset email for a test user in a menu item', () => {
```
Comment (sandewt) [inline tests/System/integration/site/components/com_users/Reset.cy.js]: send out = send

File: tests/System/integration/site/components/com_users/Reset.cy.js (hunk 5 above)
Comment (laoneo) [reply tests/System/integration/site/components/com_users/Reset.cy.js]: done, you can also do code suggestions, so we can commit them directly from the web interface ;-)

File: tests/System/integration/site/components/com_users/Reset.cy.js (hunk 6 above)
Comment (laoneo) [reply tests/System/integration/site/components/com_users/Reset.cy.js]: done

File: tests/System/integration/site/components/com_users/Reset.cy.js (hunk 5 above)
Comment (brianteeman) [reply tests/System/integration/site/components/com_users/Reset.cy.js]: no objection to the change but it was acceptable before as well

Comment (sandewt) [reply tests/System/integration/site/components/com_users/Reset.cy.js]: > done, you can also do code suggestions, so...

I have to master the code first, but it seems like a challenge to me. You still hear.

File: tests/System/integration/administrator/components/com_categories/Categories.cy.js [hunk 7]
```diff
@@ -1,4 +1,4 @@
-describe('Test that the categories back end list', () => {
+describe('Test in backend that the categories list', () => {
```
Comment (brianteeman) [inline tests/System/integration/administrator/components/com_categories/Categories.cy.js:1]:
```suggestion
describe('Test in backend that the list', () => {
```

add comment as well iof you want to explain

File: tests/System/integration/site/components/com_users/Reset.cy.js (hunk 5 above)
Comment (brianteeman) [reply tests/System/integration/site/components/com_users/Reset.cy.js]: hope this animation helps

sugg

Comment (sandewt) [reply tests/System/integration/site/components/com_users/Reset.cy.js]: Eureka

File: tests/System/integration/administrator/components/com_login/Login.cy.js [hunk 8]
```diff
@@ -1,4 +1,4 @@
-describe('Test that the login component', () => {
+describe('Test in backend that the login component', () => {
   it('can log in and out in the back end', () => {
```
Comment (sandewt) [inline tests/System/integration/administrator/components/com_login/Login.cy.js]:
```suggestion
  it('can log in and out in the backend', () => {
```
or
```suggestion
  it('can log in and out', () => {
```

**Comment by sandewt** [comment]
https://github.com/joomla/joomla-cms/blob/77fbdfdf6b00788b8bc52fea803f59b445f44dbe/tests/System/integration/site/components/com_users/Registration.cy.js#L17

Suggestion:

 it('testing registration form for **a** test user through **a** menu item', () => {

**Comment by sandewt** [comment]
`it('can show..,` and `it('can display...` are both used. 
Preference? Or leave it that way?

**Comment by laoneo** [comment]
Better to use display

File: tests/System/integration/site/components/com_users/Registration.cy.js [hunk 9]
```diff
@@ -1,11 +1,10 @@
-describe('Test the registration view', () => {
-  it('can display the registration view for test user', () => {
+describe('Test in frontend that the users registration view', () => {
+  it('can display a registration form', () => {
```
Comment (sandewt) [inline tests/System/integration/site/components/com_users/Registration.cy.js]: Suggestion: Add in line 17 an **a** (2x) too.
See comment: https://github.com/joomla/joomla-cms/pull/40386#issuecomment-1509745869

Comment (laoneo) [reply tests/System/integration/site/components/com_users/Registration.cy.js]: Rewrite the whole one as it was still not inline with the rest

**Comment by sandewt** [comment]
Thanks all

**Pull request merged**